languages/rust/exercises/concept/entry-api/.meta/design.md

```diff
@@ -0,0 +1,56 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to introduce the student to the [Entry API](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html) on `HashMap`s/`BTreeMap`s in Rust, how to use it to fetch an `Entry` from a `HashMap`, as well as use the methods on `Entry` to manipulate it.
+
+## Learning objectives
+
+Students who complete this exercise should be able to...
+
+- spot appropriate use-cases/contexts in which usage of the entry API is appropriate
+- use the entry API method `and_modify` to update a fetched `Entry`
+- use the entry API method `or_insert` to ensure an `Entry` is not `Vacant` when fetching it
+- use multiple entry API methods in conjunction with each other to update `Entry`s, such as `entry().and_modify().or_insert()`
+
+This exercise may review, but is not expected to include, other usages of `HashMap`s.
+
+## Out of scope
+
+Students who complete this exercise should not necessarily be able to...
+
+- use the entry API `insert` method, as it's currently only in nightly
+- use the entry API `or_default` method, as this is less commonly used than `or_insert`
```

Comment:
Citation needed. In particular, this exercise has the students implementing a counter class; `or_default` is particularly useful for counters.

Clippy prefers

```rust
*counter.entry(item).or_default() += 1;
```

over

```rust
*counter.entry(item).or_insert(0) += 1;
```

Reply:
Purely observationally I do see .or_insert(0) more often in Rust code (e.g. the Cookbook's examples have .or_insert(0). `rustc` itself has both variants. I can believe that clippy prefers .or_default() and the community has mostly shrugged that off, though.

What if... the community has bad style? :thinking:

Reply:
Ok, I went ahead with the suggestions and removed mentions of `and_modify` and `or_insert` from the `design.md` doc. They're now mentioned in the `after.md` doc.